### PR TITLE
cupy.cumsum:  implement its axis option

### PR DESCRIPTION
**Pull request by ShigekiKarita**
Hi,

I implement and test `cupy.cumsum` with its axis option by extending the (axis=None) version.

### CODE REVIEW

File: cupy/math/sumprod.py [hunk 1]
```diff
@@ -51,12 +51,45 @@ def prod(a, axis=None, dtype=None, out=None, keepdims=False):
...
+
+
+def _cumsum_batch(out):
+    kern = core.ElementwiseKernel(
+        'int32 pos, int32 batch', 'raw T x',
+        '''
+        int b = i % batch;
+        int j = i / batch;
+        if (j & pos) {
+          x[(A){b, j}] += x[(A){b, j ^ pos | (pos - 1)}];
```
Comment (unnonouno) [inline cupy/math/sumprod.py]: How about making two index variables simply?
```
const int dst_index[] = {b, j};
const int src_index[] = {b, j ^ pos | (pos - 1)};
```

I feel `A` is a little difficult to read.

**Comment by ShigekiKarita** [comment]
Thanks for your reviews. I have fixed them.
I'm looking forward chainer v2!;-)

File: cupy/math/sumprod.py [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 from cupy import core
 
+from six.moves import range
```
Comment (unnonouno) [inline cupy/math/sumprod.py]: Sorry don't import method directory

```
import six
...

six.moves.range(...)
```

File: tests/cupy_tests/math_tests/test_sumprod.py [hunk 3]
```diff
@@ -5,6 +5,8 @@
 import cupy
 from cupy import testing
 
+from six.moves import range
```
Comment (unnonouno) [inline tests/cupy_tests/math_tests/test_sumprod.py]: ditto

File: cupy/math/sumprod.py [hunk 4]
```diff
@@ -73,6 +109,13 @@ def cumsum(a, axis=None, dtype=None, out=None):
     else:
         out[...] = a
 
+    if axis is None:
+        out = out.ravel()
+    elif not (-a.ndim <= axis < a.ndim):
+        raise ValueError("axis(={}) out of bounds".format(axis))
```
Comment (unnonouno) [inline cupy/math/sumprod.py]: Use single quotes.

Comment (unnonouno) [inline cupy/math/sumprod.py]: Can you make a unit test for this error case using `cupy_numpy_raises`?

**Comment by ShigekiKarita** [comment]
Thanks for double check! I rewrote it.

File: cupy/math/sumprod.py [hunk 5]
```diff
@@ -2,6 +2,8 @@
 
 from cupy import core
 
+import six
```
Comment (unnonouno) [inline cupy/math/sumprod.py]: Please move this import just after `import numpy`. I mean:
```
import numpy
import six

from cupy import core
```

File: tests/cupy_tests/math_tests/test_sumprod.py [hunk 6]
```diff
@@ -5,6 +5,8 @@
 import cupy
 from cupy import testing
 
+import six
```
Comment (unnonouno) [inline tests/cupy_tests/math_tests/test_sumprod.py]: ditto

**Comment by unnonouno** [comment]
Do you have a time to fix the PR? Can i fix it?

**Comment by ShigekiKarita** [comment]
@unnonouno i fixed it

**Comment by unnonouno** [comment]
Thank you! LGTM

**Pull request merged**